cranelift-codegen/meta/src/unique_table.rs

```diff
@@ -1,15 +1,54 @@
+use std::collections::HashMap;
+use std::hash::Hash;
 use std::slice;
 
+/// Collect items into the `table` list, removing duplicates.
+pub struct UniqueTable<'entries, T: Eq + Hash> {
```

Comment:
Maybe at some point this can be replaced with something from crates.io? However I know sunfish wants to keep the dependencies down and it's pretty simple as-is..

Reply:
Yeah, it is quite simple. Now that I think about it, I even wonder why a simple BTreeMap couldn't be used (and an OrderedDict in the Python code).

Reply:
Probably IndexSet or IndexMap, I think it's quite a popular crate!

Reply:
BTreeMap required Ord on TypeSet, which they don't obviously implement; UniqueTable is a bit ugly but it's quite small, it works fine at this time and should be fast enough, so I would think it's fine to keep it (and avoid adding a new dependency).